srf_generation/Fault.py

```diff
@@ -554,8 +553,19 @@ def __init__(self, nhm_data: NHMFault):
             nhm_data.dip_dir,
         )
 
+        # Circular dependency here
+        # need the magnitude to determine the level of rounding of the length
+        # need the length to determine the magnitude
+        # Shouldn't be an issue in most cases
```

Comment:
Add a note to prospectively move this to the output stage rather than the input stage.

Reply:
Move length/magnitude calculation to the output stage?

Reply:
Ie have JIT calculation for magnitude and length?

Reply:
JIT for fault rounding. So it removes the circular dependancy.

Reply:
The length rounding is only to make SRF representation easier.